### PR TITLE
Setup Unit Testing 

### PR DESCRIPTION
**Pull request by brentwatson**
- Configured projects to use kotlin.
- Added OpenPoJo tests for DTO objects.
- Setup code coverage for repo (using codecov.io).
- Placeholder for android DTO parceling tests.

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -47,4 +61,10 @@ dependencies {
 
     // JetBrains Annotations
     compile 'org.jetbrains:annotations-java5:15.0'
+
```
Comment (msya) [inline core/build.gradle]: There is a new version of Jackson that could be used - `com.fasterxml.jackson.core:jackson-databind:2.7.1-1`.

File: android/build.gradle [hunk 2]
```diff
@@ -1,4 +1,6 @@
 apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+apply plugin: 'jacoco'
```
Comment (msya) [inline android/build.gradle]: Should `Cobertura` be used for its filtering capabilities?

File: core/build.gradle (hunk 1 above)
Comment (brentwatson) [inline core/build.gradle]: :heavy_check_mark:

File: android/build.gradle (hunk 2 above)
Comment (brentwatson) [inline android/build.gradle]: Going to stick with JaCoCo.
- Setup is easy for Java module.
- Can reuse config for android module.
- For this sdk there probably shouldn't be anything that should be ignored (at least not very much).

Comment (msya) [inline android/build.gradle]: :+1:

**Pull request merged**